httpx/client.py

```diff
@@ -580,8 +584,10 @@ def redirect_stream(
                 await response.aclose()
                 raise exc from None
             else:
+                response.history = list(history)
+                await response.aread()
                 request = next_request
-                await response.aclose()
+                history.append(response)
 
```

Comment:
As suggested by @j178, this part closely follows the behavior we have when handling redirection: copy the current history onto the latest response, then add the response to the history.

The only difference is that here we mutate `history` in-place instead of creating a new list (`history = history + [response]`) so that the change is propagated to `.send_handling_redirects()`.

It's possible to change this so that we build and maintain an `auth_history` list instead, and merge it with the initial `history` both here and `.send_handling_redirects()`, but I'm not sure the gained immutability would be worth the extra complexity.

Separate note: I don't think we test the case of redirect + auth yet, but that would be a good idea. Definitely out of scope for this PR though?